src/cli/self_update/windows.rs

```diff
@@ -437,23 +453,19 @@ mod tests {
                     .open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)
                     .unwrap();
                 let reg_value = RegValue {
-                    bytes: vec![
-                        0x00, 0xD8, // leading surrogate
-                        0x01, 0x01, // bogus trailing surrogate
-                        0x00, 0x00,
-                    ], // null
-                    vtype: RegType::REG_EXPAND_SZ,
+                    bytes: vec![0x00, 0xD8, 0x01, 0x01, 0x00, 0x00],
+                    vtype: RegType::REG_BINARY,
                 };
                 environment.set_raw_value("PATH", &reg_value).unwrap();
                 // Ok(None) signals no change to the PATH setting layer
-                fn panic(_: &str, _: String) -> Option<String> {
-                    panic!("called");
-                }
-                assert_eq!(None, super::_with_path_cargo_home_bin(panic).unwrap());
+                assert_eq!(
+                    None,
+                    super::_with_path_cargo_home_bin(|_, _| panic!("called")).unwrap()
+                );
             })
         });
         assert_eq!(
-            r"warning: the registry key HKEY_CURRENT_USER\Environment\PATH does not contain valid Unicode. Not modifying the PATH variable
+            r"warning: the registry key HKEY_CURRENT_USER\Environment\PATH is not a string. Not modifying the PATH variable
```

Comment:
I don't understand how we'll still see this error if your change to handle non-unicode values is complete.

Perhaps you're adding a new corner case, or anticipating some other sort of error?

If that non-unicode bytestring is able to trigger the error, then we don't actually handle non-unicode strings yet.
If it isn't able to trigger the error, but something else is generating the error, the test isn't clear.